src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -104,11 +106,15 @@ class CoreWorkerDirectTaskSubmitter {
                       std::pair<std::shared_ptr<WorkerLeaseInterface>, int64_t>>
       worker_to_lease_client_ GUARDED_BY(mu_);
 
-  // Whether we have a request to the Raylet to acquire a new worker in flight.
-  bool worker_request_pending_ GUARDED_BY(mu_) = false;
+  // Keeps track of scheduling classes that we have pending worker lease requests for with
+  // the raylet.
+  absl::flat_hash_set<SchedulingClass> pending_lease_requests_ GUARDED_BY(mu_);
```

Comment:
Suggest to store this in the `task_queues_` as a pair with the `std::deque`. That would avoid all the insertions/deletions from the `pending_lease_requests_` and also avoid having to clean them up separately.